src/libopensc/cwa-dnie.c

```diff
@@ -711,8 +711,10 @@ static int dnie_transmit_apdu_internal(sc_card_t * card, sc_apdu_t * apdu)
 			if (tmp == SC_APDU_CASE_3_SHORT)
 				apdu->cse = SC_APDU_CASE_4_SHORT;
 			if (apdu->resplen == 0) {	/* no response buffer: create */
-				apdu->resp = buf;
-				apdu->resplen = 2048;
+				apdu->resp = calloc(1, 2*SC_MAX_APDU_BUFFER_SIZE);
```

Comment:
You could use an internal define in `cwa-dnie.c`, for example `DNIE_MAX_INTERNAL_TRANSMIT_BYTES`...